### PR TITLE
docs: add TBD to `Lexicon.md`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Description was copy-pasted from https://lists.llvm.org/pipermail/llvm-dev/2017-September/117264.html

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
Assigning @eeckstein as someone who recently edited TBD-related files to verify that the description is correct. That old message from the mailing list is the only info on TBD I found so far.

File: docs/Lexicon.md [hunk 1]
```diff
@@ -594,6 +594,17 @@ of `Optional<Int>`.) Sugared types preserve information about the form
 and use of the type even though the behavior usually does not change
 (except for things like access control). Contrast with [canonical type](#canonical-type).
 
+## TBD
+
+Text-based dynamic library files (TBDs) are a textual representation of
+the information in a dynamic library / shared library that is required
+by the static linker - basically a symbol list of the exported symbols.
```
Comment (compnerd) [inline docs/Lexicon.md:601]:
```suggestion
by the static linker.
```
There is no reason to assume that it is just the symbol list.  There is technically additional metadata there that is inferred AIUI.

File: docs/Lexicon.md [hunk 2]
```diff
@@ -594,6 +594,17 @@ of `Optional<Int>`.) Sugared types preserve information about the form
...
+
+Apple’s SDKs originally used Mach-O Dynamic Library Stubs. Mach-O Dynamic
+Library Stubs are dynamic library files, but with all the text and data
+stripped out. TBD files were introduced to replaced Mach-O Dynamic Library
+Stub files in the SDK to further reduce its overall size.
```
Comment (compnerd) [inline docs/Lexicon.md:606]: I would drop the last sentence at least - unless you have the ability to cite this as the case.

Comment (MaxDesiatov) [inline docs/Lexicon.md:606]:
```suggestion
stripped out.
```

**Review by compnerd — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci please test and merge

**Review by CodaFi — APPROVED** [review]

**Pull request merged**